src/test/java/org/mule/tools/apikit/output/resources/ResourceGeneratorTest.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.tools.apikit.output.resources;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.commons.io.IOUtils;
+import org.junit.Test;
+import org.mule.tools.apikit.model.ApikitMainFlowContainer;
+import org.mule.tools.apikit.model.HttpListenerConfig;
+import org.mule.tools.apikit.model.HttpListenerConnection;
+import org.mule.tools.apikit.model.Properties;
+import org.mule.tools.apikit.model.ScaffolderResource;
+import org.mule.tools.apikit.model.ScaffoldingAccessories;
+import org.mule.tools.apikit.model.ScaffoldingConfiguration;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+public class ResourceGeneratorTest {
+
+  public static final String BASE_PATH = "src/test/resources/org.mule.tools.apikit.output.resources/";
+  public static final String PREFFIX_FULL = "expected-full";
+  public static final String SLASH = "/";
+
+  @Test
+  public void testNoGeneration() {
+    ScaffoldingConfiguration.Builder scaffoldingConfigurationBuilder = ScaffoldingConfiguration.builder();
+    assertNull(ResourcesGenerator.generate(scaffoldingConfigurationBuilder.build()));
+  }
+
+  @Test
+  public void testGenerationYAMLFull() throws IOException {
+    ScaffoldingConfiguration scaffoldingConfiguration = buildScaffoldingConfiguration("yaml/configuration-yaml-full.json", "123");
+    List<ScaffolderResource> generatedResources = ResourcesGenerator.generate(scaffoldingConfiguration);
+    commonAssertResources(generatedResources, 3);
+  }
+
+  @Test
+  public void testGenerationPropertiesFull() throws IOException {
+    ScaffoldingConfiguration scaffoldingConfiguration =
+        buildScaffoldingConfiguration("properties/configuration-properties-full.json", "123");
+    List<ScaffolderResource> generatedResources = ResourcesGenerator.generate(scaffoldingConfiguration);
+    commonAssertResources(generatedResources, 3);
+  }
```

Comment:
Despite the fact that you created some tests by generating the configuration manually, I would expect more tests on the different variants that the developer might write down the JSON and also some assertions on what the output of the generator would be.